lib/trace.js

```diff
@@ -262,42 +268,43 @@ const _resolveDep = async ({
       // ```
       //
       // The **root** one (`ROOT/nested/from-root.js`) always wins.
-      if (!pkgPaths.length) {
+      if (!pkgPaths.length && !isRelative) {
         const pkgDir = path.dirname(pkgfile);
+
+        // Special case handle "passthrough" export configuration to make
+        // modern ESM act like old file-based CommonJS.
+        //
+        // Our approach is to simply ignore these declarations since the
+        // `resolve` library takes care of the straight path-based file
+        // resolution already and there's nothing _different_ we expect
+        // to happen with an export mapping.
+        //
+        // **NOTE**: Mutates `package.json`.
+        //
+        // From: https://nodejs.org/api/packages.html#packages_package_entry_points
+        // > As a last resort, package encapsulation can be disabled entirely
+        // > by creating an export for the root of the package "./*": "./*".
+        // > This exposes every file in the package at the cost of disabling
+        // > the encapsulation and potential tooling benefits this provides.
+        // > As the ES Module loader in Node.js enforces the use of the full
+        // > specifier path, exporting the root rather than being explicit
+        // > about entry is less expressive than either of the prior examples.
+        // > Not only is encapsulation lost but module consumers are unable to
+        // > import feature from 'my-mod/feature' as they need to provide the
+        // > full path import feature from 'my-mod/feature/index.js.
+        Object.entries(PASSTHROUGH_EXPORTS).forEach(([key, vals]) => {
```

Comment:
This should have originally been down before looping on CONDITIONS and not repeatedly in CONDITIONS.